test/unit/modules/helpers/get-n-parents-spec.js

```diff
@@ -28,7 +27,113 @@ describe('RecordingLibrary.helpers.getNParents', () => {
   p2.appendChild(p1);
   p1.appendChild(node);
 
-  test('should return element attributes up 5 parents away if no limit is specified', () => {
+  test(`should return element attributes up 5 parents away if
+    (1) no limit is specified
+    (2) tagName does not include 'button', 'select', 'input', or 'a'
+    (3) role attribute does not include 'button', 'checkbox', 'radio', 'link'`,
+  () => {
+    expect(getNParents(node)).toEqual([
+      {
+        class: 'span-class',
+        id: null,
+        name: null,
+        all_attrs: {
+          class: 'span-class',
+          unique_selector: '.span-class',
+        },
+        node_name: 'SPAN',
+        tag_name: 'SPAN',
+        text: null,
+        title: null,
+        type: null,
+        href: null,
+        nth_parent: 1,
+      },
+      {
+        class: null,
+        id: 'pid',
+        name: null,
+        all_attrs: {
+          id: 'pid',
+          unique_selector: '#pid',
+        },
+        node_name: 'P',
+        tag_name: 'P',
+        text: null,
+        title: null,
+        type: null,
+        href: null,
+        nth_parent: 2,
+      },
+      {
+        class: null,
+        id: null,
+        name: 'divname',
+        all_attrs: {
+          name: 'divname',
+          unique_selector: 'div',
+        },
+        node_name: 'DIV',
+        tag_name: 'DIV',
+        text: null,
+        title: null,
+        type: null,
+        href: null,
+        nth_parent: 3,
+      },
+      {
+        class: 'label-class',
+        id: null,
+        name: null,
+        all_attrs: {
+          class: 'label-class',
+          unique_selector: '.label-class',
+        },
+        node_name: 'LABEL',
+        tag_name: 'LABEL',
+        text: null,
+        title: null,
+        type: null,
+        href: null,
+        nth_parent: 4,
+      },
+      {
+        class: null,
+        id: null,
+        name: null,
+        all_attrs: {
+          title: 'h1title',
+          unique_selector: 'h1',
+        },
+        node_name: 'H1',
+        tag_name: 'H1',
+        text: null,
+        title: 'h1title',
+        type: null,
+        href: null,
+        nth_parent: 5,
+      },
+    ]);
+  });
+
+  test(`should return element attributes up 9 parents away instead of 5 parents away because:
```

Comment:
[not blocking] Huh, I thought we were only intending to grab a total of 7 🤔 But looking at the [code](https://github.com/ProdPerfect/prodperfect-keen-tracking.js/pull/78/files), we actually don't put an upper bound on it.

Reply:
yea, I think we just keep looping and looping and looping so if there are two buttons or a button and an anchor tag... lolz